### PR TITLE
Added extended endpoint for vehicles

### LINKED ISSUE #85: Extend the /vehicles endpoint
**Describe the feature**
Add endpoints like **GET**  `vehicles/2`, 
Add endpoints like **GET** `vehicles?manufacturer=tata` .
`vehicles/2` will return 2 index record and `vehicles?manufacturer=tata`  will return manufacturer tata records from the JSON.

**Are you working on this feature? (Yes/No)**
Yes

### PR DESCRIPTION
**Pull request by Samridhi-98**
# Related Issue

-   closes #85 

# Proposed Changes

-  Added endpoint `/vehicles/index` in **vehicles.ts**
-  Added endpoint `/vehicles?manufacturer=Tesla` in **vehicles.ts**

# Checklist

-   [x] Tests
-   [x] Translations

### CODE REVIEW

**Comment by Samridhi-98** [comment]
@paramsiddharth @ShauryaAg  Please review my PR.

File: src/api/routes/vehicles.ts [hunk 1]
```diff
@@ -8,7 +8,33 @@ const router = express.Router();
...
+        }
+        else {
+            res.status(200).json(data);
+        }
+    } catch (err) {
+        res.status(500).json({ error: err });
+    }
+});
+
+router.get("/:index", (req: Request, res: Response, next: NextFunction) => {
```
Comment (ShauryaAg) [inline src/api/routes/vehicles.ts:29]: Please change `index` to `id`

File: src/api/routes/vehicles.ts [hunk 2]
```diff
@@ -8,7 +8,33 @@ const router = express.Router();
...
+        if (manufacturer !== undefined) {
+            for (let i: any = 0; i < data.data.length; i++) {
+                if (data.data[i].manufacturer === manufacturer) {
+                    result.push(data.data[i]);
+                }
+            }
+            res.status(200).json({ data: result });
+        }
+        else {
+            res.status(200).json(data);
```
Comment (ShauryaAg) [inline src/api/routes/vehicles.ts:22]: Please return `400` if `query.manufacturer` does not match any given option.

**Review by ShauryaAg — CHANGES_REQUESTED** [review]

File: src/api/routes/vehicles.ts (hunk 2 above)
Comment (Samridhi-98) [reply src/api/routes/vehicles.ts:22]: Isn't it wrong? 
Since its `/ ` route so if the user wont specify any query it should return the whole data.

Comment (ShauryaAg) [reply src/api/routes/vehicles.ts:22]: Say, the user hits `/vehicles?manufacturer=qwerty12345` endpoint, which is not a manufacturer of any of the available cars. Then it should return `400`

**Comment by Samridhi-98** [comment]
@ShauryaAg  Done!

**Review by paramsiddharth — APPROVED** [review]

**Review by ShauryaAg — APPROVED** [review]

**Comment by paramsiddharth** [comment]
Thank you!

**Pull request merged**